### ISSUE TITLE
TweetTokenizer quirk with long numbers

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#1799]
When there's a number token that's >8 length, the `TweetTokenizer` tries to iteratively parse the first 8-10 digits as a phone number if it matches the phone number regex on https://github.com/alvations/nltk/blob/develop/nltk/tokenize/casual.py#L125

Maybe, there should be an option to allow users to keep long digits and not try to recognize the first part as a number. 

Or better yet, if there's regex fix that can prevent the number from getting split if it's longer than 10. 

Details on https://stackoverflow.com/questions/45425946/tokenize-in-nltk-tweettokenizer-returning-integers-by-splitting/45431922#45431922

**Comment by ali-abdullah** [comment]
Edit- After reading the stackoverflow discussion I realise I might have missed and repeated some points, please let me know if I have overlooked something

## TL;DR
I suggest adding a flag to check if the user wants to recognise large numbers and not phone numbers or add ` \d*` to the regex to include numbers greater then 8. 

The second solution might result in some weird cases, that shouldn't pop up to start with. 

Still new to tokenisers and nlp as a whole, so any suggestions or constructive criticism would be appreciated. 

## Problem
So due to the bug caused by the Regex, numbers of length 8, 9, 10 and greater that don't begin with a 1 will be split as below 

_tw = TweetTokenizer()_

**8 digits:**
_tw.tokenize('23456789') = [u'2345678', u'9']_

**9 digits:**
_tw.tokenize('234567890')  = [u'2345678', u'90']_

**10 digits:**
_tw.tokenize('2345678902') = [u'2345678902']_

**Greater than 10 (splits after every 10 digits):**
_tw.tokenize('234567890223342') = [u'2345678902', u'23342']_

If the number 1 is present at the start of any number or split, it's added to the beginning and doesn't effect the rest of the output, so for example:

**9 digits example now has 1 at the start, so it splits into 9 digits and 1 digit**
_tw.tokenize('123456789') = [u'12345678', u'9']_

## Solutions
Looking at the problem we could solve the issue in two ways:

1. editing the Regex by adding `\d*` at the end (please look at the code snippet), hence if any more numbers are added after 7, i.e 8,9,10 or greater, they will simply be considered as one token.

**9 digits:**
_tw.tokenize('234567890')  = [u'234567890']_

**Greater than 10 (splits after every 10 digits):**
_tw.tokenize('234567890223342') = [u'234567890223342']_

This solution could cause awkward situation for example:
_tw.tokenize('+1-(234)-567-890123456') = [u'+1-(234)-567-890123456']_  

However, at this point the phone number itself is incorrect and has too many digits.

```
# The components of the tokenizer:
REGEXPS = (
    URLS,
    # Phone numbers:
    r"""
    (?:
      (?:            # (international)
        \+?[01]
        [\-\s.]*
      )?
      (?:            # (area code)
        [\(]?
        \d{3}
        [\-\s.\)]*
      )?
      \d{3}          # exchange
      [\-\s.]*
      \d{4}          # base
      \d*            # incase of extra numbers consider the whole thing to be  a number > 8
    )"""
    ,
```

2.  The second option is creating an option that the user can set true or false, depending on whether they want to keep large numbers and not look tokenise phone numbers using this particular regex, in which case you could remove the REGEXPS  index for the above Phone number regex when the user sets the flag to true.

```
# This is the core tokenizing regex:

WORD_RE = re.compile(r"""(%s)""" % "|".join(REGEXPS), re.VERBOSE | re.I
                     | re.UNICODE)
```

**Comment by ali-abdullah** [comment]
hey @alvations , I can apply the first fix (\d*) if you think that's alright.
Just wanted to make sure before submitting a pull request

**Comment by alvations** [comment]
@ali-abdullah sorry for getting back to you this late. I'm not very familiar with the twitter tokenizer though. 

Please do feel free to submit a pull request and someone will review it and see whether the `\d*` fix is okay.
